proof_ipa.go

```diff
@@ -257,32 +292,35 @@ func SerializeProof(proof *Proof) (*VerkleProof, StateDiff, error) {
 			stemdiff = &statediff[len(statediff)-1]
 			copy(stemdiff.Stem[:], stem)
 		}
-		stemdiff.SuffixDiffs = append(stemdiff.SuffixDiffs, SuffixStateDiff{Suffix: key[StemSize]})
-		newsd := &stemdiff.SuffixDiffs[len(stemdiff.SuffixDiffs)-1]
-
-		var valueLen = len(proof.PreValues[i])
-		switch valueLen {
-		case 0:
-			// null value
-		case 32:
-			newsd.CurrentValue = (*[32]byte)(proof.PreValues[i])
+		preLen := len(proof.PreValues[i])
+		postLen := len(proof.PostValues[i])
+		switch {
+		case isInsertion(preLen, postLen):
+			var newValue [32]byte
+			copy(newValue[:], proof.PostValues[i])
```

Comment:
Note that we don't have to do the `postLen == 32` and all the alignment thing.

If `proof.PostValues[i]` isn't 32 bytes, then `copy` will copy whatever amount of bytes there are. So both in aligned and unaligned case, the right thing will happen.

Same applies to other cases.